File: app/src/main/java/com/se/music/activity/CreateSongListActivity.java
package com.se.music.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.widget.EditText;
import android.widget.TextView;

import com.se.music.database.provider.SongListDBService;
import com.se.music.pastmusic.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class CreateSongListActivity extends AppCompatActivity {

    String listName;
    String listInfo;

    @BindView(R.id.et_list_name)
    EditText nameInput;

    @BindView(R.id.et_list_info)
    EditText infoInput;

    @OnClick(R.id.back)
    void back() {
        finish();
    }

    @BindView(R.id.title)
    TextView mTitle;

    @OnClick(R.id.save)
    void save() {
        listName = nameInput.getText().toString();
        listInfo = infoInput.getText().toString();
        if (!TextUtils.isEmpty(listName)) {
            SongListDBService.Companion.getInstance().insert(listName, listInfo);
        }
        finish();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_create_song_list);
        setStatusBar();
        ButterKnife.bind(this);

        mTitle.setText("新建歌单");
        nameInput.setText("新建歌单");
        nameInput.setSelectAllOnFocus(true);
        nameInput.requestFocus();
    }

    protected void setStatusBar() {
    }

    @Override
    public void finish() {
        super.finish();
        this.overridePendingTransition(R.anim.push_up_in, R.anim.push_down_out);
    }
}

File: app/src/main/kotlin/com/se/music/activity/CreateSongListActivity.kt
package com.se.music.activity

import android.content.ContentValues
import android.os.Bundle
import android.view.*
import android.widget.EditText
import com.se.music.R
import com.se.music.base.ToolBarActivity
import com.se.music.provider.metadata.*
import com.se.music.utils.generateLoaderId
import java.util.*

/**
 *Author: gaojin
 *Time: 2018/5/13 下午5:49
 */

class CreateSongListActivity : ToolBarActivity() {

    companion object {
        val resultCode = generateLoaderId()
    }

    private lateinit var nameInput: EditText
    private lateinit var infoInput: EditText

    override fun createContentView(inflater: LayoutInflater, rootView: ViewGroup): View {
        return inflater.inflate(R.layout.activity_create_song_list, rootView, false)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        title = resources.getString(R.string.create_new_song_list)
        hideStatStatusBar()
        nameInput = findViewById(R.id.et_list_name)
        infoInput = findViewById(R.id.et_list_info)
        nameInput.setText(resources.getString(R.string.create_new_song_list))
        nameInput.setSelectAllOnFocus(true)
        nameInput.requestFocus()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        super.onCreateOptionsMenu(menu)
        menuInflater.inflate(R.menu.menu_save, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            R.id.menu_save -> save()
        }
        return super.onOptionsItemSelected(item)
    }

    fun save() {
        val listName = nameInput.text.toString()
        val listInfo = infoInput.text.toString()
        val values = ContentValues()
        values.run {
            put(SL_ID, UUID.randomUUID().toString())
            put(SL_NAME, listName)
            put(SL_CREATE_TIME, System.currentTimeMillis())
            put(SL_INFO, listInfo)
        }
        contentResolver.insert(SL_CONTENT_URI, values)
        setResult(resultCode)
        finish()
    }

    override fun finish() {
        super.finish()
        overridePendingTransition(R.anim.push_up_in, R.anim.push_down_out)
    }

}